webapi/Controllers/PluginController.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using CopilotChat.WebApi.Auth;
+using CopilotChat.WebApi.Hubs;
+using CopilotChat.WebApi.Models.Storage;
+using CopilotChat.WebApi.Options;
+using CopilotChat.WebApi.Storage;
+using CopilotChat.WebApi.Utilities;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.Diagnostics;
+
+namespace CopilotChat.WebApi.Controllers;
+
+/// <summary>
+/// Controller responsible for returning the service options to the client.
+/// </summary>
+[ApiController]
+public class PluginController : ControllerBase
+{
+    private const string PluginStateChanged = "PluginStateChanged";
+    private readonly ILogger<PluginController> _logger;
+    private readonly IDictionary<string, Plugin> _availablePlugins;
+    private readonly ChatSessionRepository _sessionRepository;
+
+    public PluginController(
+        ILogger<PluginController> logger,
+        IDictionary<string, Plugin> availablePlugins,
+        ChatSessionRepository sessionRepository)
+    {
+        this._logger = logger;
+        this._availablePlugins = availablePlugins;
+        this._sessionRepository = sessionRepository;
+    }
+
+    /// <summary>
+    /// Fetches a plugin's manifest.
+    /// </summary>
+    /// <param name="manifestDomain">The domain of the manifest.</param>
+    /// <returns>The plugin's manifest JSON.</returns>
+    [HttpGet]
+    [Route("getPluginManifest")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    public async Task<IActionResult> GetPluginManifest([FromQuery] Uri manifestDomain)
+    {
+        using var request = new HttpRequestMessage(HttpMethod.Get, PluginUtils.GetPluginManifestUri(manifestDomain));
+        // Need to set the user agent to avoid 403s from some sites.
+        request.Headers.Add("User-Agent", Telemetry.HttpUserAgent);
+
+        using HttpClient client = new();
+        var response = await client.SendAsync(request);
+        if (!response.IsSuccessStatusCode)
+        {
+            return this.StatusCode((int)response.StatusCode, await response.Content.ReadAsStringAsync());
+        }
+
+        return this.Ok(await response.Content.ReadAsStringAsync());
+    }
+
+    /// <summary>
+    /// Enable or disable a plugin for a chat session.
+    /// </summary>
+    [HttpPut]
+    [ActionName("SetPluginStateAsync")]
```

Comment:
To make this route RESTfull, I think it should be modeled as a mutation of a resource, either the plugin resource or the chat resource. So either `PUT plugins/{pluginId}/chats/{chatId}` or `PUT chats/{chatId}/plugins/{pluginId}`